records_mover/records/schema/field/__init__.py

```diff
@@ -192,40 +184,41 @@ def components_to_time_str(df: pd.DataFrame) -> datetime.time:
                         return datetime.time(hour=df['hours'],
                                              minute=df['minutes'],
                                              second=df['seconds'])
+                    logger.debug("Applying pd.Timedelta logic on series for %s", self.name)
                     out = series.dt.components.apply(axis=1, func=components_to_time_str)
                     return out
 
-        return series.astype(self.to_numpy_dtype())
+        target_type = self.to_pandas_dtype()
+        logger.debug("Casting field %s from type %r to type %s", self.name, series.dtype,
+                     target_type)
+        return series.astype(target_type)
 
-    def to_numpy_dtype(self) -> Union[Type[Any], str]:
+    def to_pandas_dtype(self) -> Union[Type[Any], str]:
         import numpy as np
+        import pandas as pd
+        from .pandas import supports_nullable_ints, integer_type_for_range
+
+        has_extension_types = supports_nullable_ints()
 
         if self.field_type == 'integer':
             int_constraints =\
                 cast(Optional[RecordsSchemaFieldIntegerConstraints], self.constraints)
             min_: Optional[int] = None
             max_: Optional[int] = None
+            required = False
             if int_constraints:
                 min_ = int_constraints.min_
                 max_ = int_constraints.max_
+                required = int_constraints.required
+
+            if not required and not has_extension_types:
+                logger.warning(f"Dataframe field {self.name} is nullable, but using pandas "
+                               f"{pd.__version__} which does not support nullable integer type")
 
             if min_ is not None and max_ is not None:
-                if min_ >= INT8_MIN and max_ <= INT8_MAX:
-                    return np.int8
-                elif min_ >= UINT8_MIN and max_ <= UINT8_MAX:
-                    return np.uint8
-                elif min_ >= INT16_MIN and max_ <= INT16_MAX:
-                    return np.int16
-                elif min_ >= UINT16_MIN and max_ <= UINT16_MAX:
-                    return np.uint16
-                elif min_ >= INT32_MIN and max_ <= INT32_MAX:
-                    return np.int32
-                elif min_ >= UINT32_MIN and max_ <= UINT32_MAX:
-                    return np.uint32
-                elif min_ >= INT64_MIN and max_ <= INT64_MAX:
-                    return np.int64
-                elif min_ >= UINT64_MIN and max_ <= UINT64_MAX:
-                    return np.uint64
+                typ = integer_type_for_range(min_, max_, has_extension_types)
+                if typ:
+                    return typ
```

Comment:
Small thing, but I'd recommend keeping to one single style for avoiding keywords (either dropping letters or adding an underscore after) - there's two used in this function.